### ISSUE TITLE
v14, async error messages no longer useful

### ISSUE THREAD

**Issue by garyking** [yargs/yargs#1419]
Since upgrading to v14, async error messages aren't useful anymore.

Previously, I would get a full stack trace of my async error. Now, I get the following error:

```
(node:98953) UnhandledPromiseRejectionWarning: TypeError: Cannot convert undefined or null to object
    at Function.keys (<anonymous>)
    at Object.help (node_modules/yargs/lib/usage.js:240:22)
    at Object.self.showHelp (node_modules/yargs/lib/usage.js:432:15)
    at Object.Yargs.self.showHelp (node_modules/yargs/yargs.js:790:11)
    at Object.fail (node_modules/yargs/lib/usage.js:50:17)
    at node_modules/yargs/lib/command.js:239:36
    at processTicksAndRejections (internal/process/task_queues.js:85:5)
(node:98953) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:98953) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

My Yargs code:

```js
require('yargs')
  .command(
    'command',
    'Desc',
    () => {},
    async () => {
      await commands.run()
    },
  )
  .alias('h', 'help')
  .demandCommand()
  .strict()
  .version(false)
  .wrap(null)
  .parse();
```

Is it something to do with `parse()` that I have to change for v14?

**Comment by garyking** [comment]
Okay, so I see that I gotta basically replace `.parse()` with `.argv`. But now ESLint throws `no-unused-expressions`. I know I can disable this error, but is there another way around it, with Yargs?

**Comment by mleguen** [comment]
About your original issue: this is probably related to something @bcoe corrected in #1412 (another exception in yargs masking your own exception). Could you please try your code with the `master` branch of yargs to see if it is corrected?

**Comment by mleguen** [comment]
To deal with your linting issue, could you please provide your eslint configuration?

I tried with the default eslint configuration (letting eslint detect the standard to use from the code you provided, which indeed sets `no-unused-expressions: error` in `.eslintrc.yml`) but it did not report any error.

**Comment by garyking** [comment]
I can confirm that the problem is fixed in the Yargs master branch. Please release it soon, thanks.

---

The following in a `.eslintrc.js` is enough to throw `no-unused-expressions`.

```js
module.exports = {
  extends: ['airbnb'],
};
```

**Comment by bcoe** [comment]
@garyking @mleguen 👋 will work on getting a release out soon (I've been a bit heads down working on some features for Node.js, sorry for the slow turnaround).

**Comment by garyking** [comment]
This issue is fixed in Yargs 14.1.0, but that version is tagged as `next` rather than `latest`. Is this intentional?

**Comment by garyking** [comment]
Seems good now, with 14.2.0 being out and tagged as `latest`.
